### PR TITLE
Remove DRAM node of height 0 skiplist data

### PR DESCRIPTION
**Pull request by JiayuZzz**
- Add offset type in hash entry to indicate what it's pointing to
- Remove DRAM node of height 0 skiplist data, and directly store its data entry offset in hash entry

### CODE REVIEW

**Comment by JiayuZzz** [comment]
Benchmark: Saved  13GB memory for 100GB data

**Comment by JiayuZzz** [comment]
@KVKIT @ZiyanShi @sherry-1001

File: engine/hash_table.hpp [hunk 1]
```diff
@@ -29,16 +29,29 @@ enum class HashEntryStatus : uint16_t {
...
+  // Offset is pointer to a skiplist node
+  SkiplistNode = 3,
+  // Offset is pointer to a skiplist struct
+  Skiplist = 4,
+};
+
 struct HashHeader {
   uint32_t key_prefix;
-  uint16_t type;
+  uint16_t data_type;
```
Comment (ZiyanShi) [inline engine/hash_table.hpp:45]: please don't... for now.

File: engine/data_entry.hpp [hunk 2]
```diff
@@ -29,6 +29,9 @@ enum DATA_ENTRY_TYPE : uint16_t {
   PADDING = 1 << 15,
 };
 
+const uint16_t SortedDataEntryType =
```
Comment (sherry-1001) [inline engine/data_entry.hpp:32]: Where used ？

Comment (JiayuZzz) [reply engine/data_entry.hpp:32]: just a placeholder

**Pull request merged**